### PR TITLE
Add Contacts Sample to Template App

### PR DESCRIPTION
**Pull request by kchitalia**
@smcnulty-sfdc

### CODE REVIEW

File: SalesforceSDK/Salesforce.Universal.Template/Salesforce.Universal.Template.Shared/Pages/MainPage.xaml.cs [hunk 1]
```diff
@@ -53,11 +57,41 @@ namespace $safeprojectname$.Pages
     /// </summary>
     public sealed partial class MainPage : NativeMainPage
     {
+        private const string ApiVersion = "v31.0";
```
Comment (bhariharan) [inline SalesforceSDK/Salesforce.Universal.Template/Salesforce.Universal.Template.Shared/Pages/MainPage.xaml.cs]: I'd recommend creating an ```ApiUtils``` class in the Mobile SDK library itself to store this constant. Makes updates to API version easier to do, as opposed to having to change it in each sample app.

File: SalesforceSDK/Salesforce.Universal.Template/Salesforce.Universal.Template.Shared/Pages/MainPage.xaml.cs [hunk 2]
```diff
@@ -53,11 +57,41 @@ namespace $safeprojectname$.Pages
...
         }
+
+        protected override async void OnNavigatedTo(NavigationEventArgs e)
+        {
+            Account account = AccountManager.GetAccount();
+            if (account != null)
+            {
+                try
+                {
+                    account = await OAuth2.RefresAuthToken(account);
```
Comment (bhariharan) [inline SalesforceSDK/Salesforce.Universal.Template/Salesforce.Universal.Template.Shared/Pages/MainPage.xaml.cs]: Should we rename this to ```RefreshAuthToken```? There's an ```h``` missing :-)

**Comment by bhariharan** [comment]
Can we also bump up the version number of the library to ```3.3.0.unstable``` in the ```unstable``` branch?

File: SalesforceSDK/Salesforce.Universal.Template/Salesforce.Universal.Template.Shared/Pages/MainPage.xaml.cs (hunk 1 above)
Comment (kchitalia) [inline SalesforceSDK/Salesforce.Universal.Template/Salesforce.Universal.Template.Shared/Pages/MainPage.xaml.cs]: Actually there is a ApiVersionStrings.VersionNumber that does this. I'll change it to use that

File: SalesforceSDK/Salesforce.Universal.Template/Salesforce.Universal.Template.Shared/Pages/MainPage.xaml.cs (hunk 2 above)
Comment (kchitalia) [inline SalesforceSDK/Salesforce.Universal.Template/Salesforce.Universal.Template.Shared/Pages/MainPage.xaml.cs]: Oh yes, thanks for catching that

**Comment by kchitalia** [comment]
Yep, I will send in a new pull request for that

**Pull request merged**